xmtp_mls/src/groups/message_history.rs

```diff
@@ -182,7 +184,45 @@ where
         Ok(())
     }
 
-    pub(crate) async fn prepare_groups_to_sync(&self) -> Result<Vec<StoredGroup>, StorageError> {
+    async fn prepare_history_reply(
+        &self,
+        request_id: &str,
+        url: &str,
+    ) -> Result<HistoryReply, MessageHistoryError> {
+        let (history_file, enc_key) = self.write_history_bundle().await?;
+
+        let signing_key = HistoryKeyType::new_chacha20_poly1305_key();
```

Comment:
We should use a different function name for this if it's not actually returning a chacha20 key used for encryption

Reply:
fixed to actually use the `chacha20` cipher in [9896f89](https://github.com/xmtp/libxmtp/pull/791/commits/9896f892ae2d30776d9b170635c5bbafe32c12cb)

Reply:
It doesn't really matter since both are just a bunch of random bytes, but what I was getting at is that you aren't using it as a chacha20 key. You're using it as a MAC key.